keras_hub/src/models/object_detector_preprocessor.py

```diff
@@ -53,5 +58,5 @@ def __init__(
     @preprocessing_function
     def call(self, x, y=None, sample_weight=None):
         if self.image_converter:
-            x = self.image_converter(x)
+            x, y, sample_weight = self.image_converter(x, y, sample_weight)
```

Comment:
@mattdangerw 

Here we pass the x -> image, y-> {"boxes": <>, "labels": <>}

While prediction x-> images.

And later in image_converter we form the dictionary like:

{"image" x, "bounding_boxes": y} -> this is sent to Resizing layer while training and boxes are properly transformed.

Prediction we just directly pass x to resizing. Keras layer takes care of the just images so we dont dictionary scenario here.

Test works for both the cases.